core/src/main/java/org/infinispan/filter/CompositeKeyFilter.java

```diff
@@ -19,4 +30,43 @@ public boolean accept(K key) {
          if (!k.accept(key)) return false;
       return true;
    }
+
+   @Inject
+   protected void injectDependencies(ComponentRegistry cr) {
+      for (KeyFilter<? super K> f : filters) {
+         cr.wireDependencies(f);
+      }
+   }
+
+   public static class Externalizer extends AbstractExternalizer<CompositeKeyFilter> {
+
+      @Override
+      public Set<Class<? extends CompositeKeyFilter>> getTypeClasses() {
+         return Util.<Class<? extends CompositeKeyFilter>>asSet(CompositeKeyFilter.class);
+      }
+
+      @Override
+      public void writeObject(ObjectOutput output, CompositeKeyFilter object) throws IOException {
+         output.writeInt(object.filters.length);
```

Comment:
We should be able to use the writeUnsignedInt method from UnsignedNumeric class.